### PR TITLE
update tests for windows

### PR DESCRIPTION
**Pull request by aramase**
- Updates makefile to add rule for windows-e2e
- Adds helm chart for windows

fixes https://github.com/kubernetes-sigs/secrets-store-csi-driver/issues/179

### CODE REVIEW

**Comment by aramase** [comment]
/hold

(using prow to test the PR as having trouble with dev machine). Will remove hold and assign when ready for review

File: charts/secrets-store-csi-driver/templates/csidriver.yaml [hunk 1]
```diff
@@ -5,7 +5,8 @@ metadata:
 spec:
   podInfoOnMount: true
   attachRequired: false
-{{- if and ( ge (.Capabilities.KubeVersion.Major | int) 1 ) ( ge (.Capabilities.KubeVersion.Minor | int) 16 ) }}
-  volumeLifecycleModes: # Added in Kubernetes 1.16 with default mode of Persistent. Secrets store csi driver needs Ephermeral to be set.
+{{- if semverCompare ">=1.16-0" .Capabilities.KubeVersion.GitVersion }}
```
Comment (aramase) [inline charts/secrets-store-csi-driver/templates/csidriver.yaml:8]: Using int compare does not work for pre-release versions (alpha/beta). So switched this to use `semverCompare` instead.

File: charts/secrets-store-csi-driver/templates/secrets-store-csi-driver.yaml [hunk 2]
```diff
@@ -1,3 +1,4 @@
+{{- if eq .Values.windows.enabled false }}
```
Comment (ritazh) [inline charts/secrets-store-csi-driver/templates/secrets-store-csi-driver.yaml:1]: What about hybrid clusters?

File: Makefile [hunk 3]
```diff
@@ -65,37 +68,76 @@ endif
...
+	curl https://raw.githubusercontent.com/helm/helm/master/scripts/get-helm-3 | bash
+
+.PHONY: e2e-teardown
+e2e-teardown:
+	helm delete csi-secrets-store --namespace default
+
+.PHONY: install-driver
+install-driver:
+ifdef TEST_WINDOWS
+		helm install csi-secrets-store charts/secrets-store-csi-driver --namespace default --wait --timeout=15m -v=5 --debug \
```
Comment (ritazh) [inline Makefile:124]: maybe use the same helm install cmd but set diff variables in the if/else?

File: charts/secrets-store-csi-driver/templates/secrets-store-csi-driver.yaml (hunk 2 above)
Comment (aramase) [reply charts/secrets-store-csi-driver/templates/secrets-store-csi-driver.yaml]: Updated to support hybrid scenarios

File: Makefile (hunk 3 above)
Comment (aramase) [reply Makefile:124]: The variables for linux and windows are different which is why the helm install cmd is different in if/else

**Comment by aramase** [comment]
@ritazh PTAL when you get a chance! Updated the charts to support hybrid scenarios and also added liveness probe sidecar for windows.

File: charts/secrets-store-csi-driver/values.yaml [hunk 4]
```diff
@@ -1,7 +1,16 @@
...
+    repository: docker.io/deislabs/secrets-store-csi
+    tag: v0.0.9
+    pullPolicy: Always
+
+windows:
+  enabled: false
+  image:
+    repository: mcr.microsoft.com/k8s/csi/secrets-store/driver
+    tag: v0.0.9
+    pullPolicy: IfNotPresent
```
Comment (ritazh) [inline charts/secrets-store-csi-driver/values.yaml:13]: Why is this `IfNotPresent`

Comment (aramase) [reply charts/secrets-store-csi-driver/values.yaml:13]: In general, windows images could be large and we want to prevent pulling image `Always`. wdyt?

Comment (ritazh) [reply charts/secrets-store-csi-driver/values.yaml:13]: yea that makes sense

**Pull request merged**